### PR TITLE
sys var: add `tidb_enable_shared_lock_promotion` related doc change

### PR DESCRIPTION
**Pull request by cfzjywxk**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Referenced tidb issue https://github.com/pingcap/tidb/issues/55022, the document for supporting upgrading `for share` lock statement to `for update` statement.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -3312,6 +3312,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`60s`
 - 刚重启的 TiFlash 可能不能正常提供服务。为了防止查询失败，TiDB 会限制 tidb-server 向刚重启的 TiFlash 节点发送查询。这个变量表示刚重启的 TiFlash 不被发送请求的时间范围。
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。注意该变量不可和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启。
```
Comment (joey-yez) [inline system-variables.md:3321]: @cfzjywxk 如果同时开启 enable noop function，是不是shared lock 应该以这个变量为准？

Comment (cfzjywxk) [reply system-variables.md]: @joey-yez 目前如果开了 enable noop function 再开 shared lock upgrade，会报错。在设置 variable 那里有检查

Comment (cfzjywxk) [reply system-variables.md]: 描述已修改

File: system-variables.md [hunk 2]
```diff
@@ -2291,6 +2291,14 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 可选值：`OFF`，`ON`
 - 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON`，则优先使用缓存中的 Chunk 对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF`，则直接从系统内存中申请 Chunk 对象。
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。注意该变量不可和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启。
```
Comment (hfxsd) [inline system-variables.md:2300]:
```suggestion
- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，会覆盖 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 的设置。
```

File: system-variables.md [hunk 3]
```diff
@@ -3312,6 +3320,18 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，会覆盖 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 的设置。
+
+>>>>>>> a8de92e88629f34c616bd02f2c9bbb2cf6fa085d
```
Comment (hfxsd) [inline system-variables.md:3334]:
```suggestion
```

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -107,7 +107,9 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE``。
```
Comment (hfxsd) [inline pessimistic-transaction.md:110]:
```suggestion
    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`。
```

File: system-variables.md [hunk 5]
```diff
@@ -2291,6 +2291,15 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 可选值：`OFF`，`ON`
 - 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON`，则优先使用缓存中的 Chunk 对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF`，则直接从系统内存中申请 Chunk 对象。
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
```
Comment (hfxsd) [inline system-variables.md:2297]:
```suggestion
- 是否持久化到集群：是
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
```

File: system-variables.md [hunk 6]
```diff
@@ -2291,6 +2291,15 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，会覆盖 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 的设置。
```
Comment (hfxsd) [inline system-variables.md:2301]:
```suggestion
- 该变量用于控制是否启用共享锁升级为排他锁的功能。当该变量值为 `ON` 时，TiDB 会尝试将共享锁升级为排他锁，以减少死锁的概率。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
- 如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，会覆盖 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 的设置。
```

File: system-variables.md [hunk 7]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`, 当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+- 如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，会覆盖 `tidb_enable_noop_functions` 的设置。
```
Comment (joey-yez) [inline system-variables.md:2302]:
```suggestion
- 如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，对于 `SELECT LOCK IN SHARE MODE` 语句 TiDB 会优先启用该功能。
```

**Review by joey-yez — APPROVED** [review]

File: system-variables.md [hunk 8]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`, 当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+- 如该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，对于 `SELECT LOCK IN SHARE MODE` 语句 TiDB 会优先启用该功能。
```
Comment (hfxsd) [inline system-variables.md:2302]:
```suggestion
- 如果该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，该变量会对 `SELECT LOCK IN SHARE MODE` 语句生效。
```

File: pessimistic-transaction.md [hunk 9]
```diff
@@ -107,7 +107,9 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`。
+
+    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_upgrade`](/system-variables.md#tidb_enable_shared_lock_upgrade-span-classversion-mark从-v830-版本开始引入span) 系统变量开启 `SELECT LOCK IN SHARE MODE` 语句的功能。开启后，`SELECT LOCK IN SHARE MODE` 语句会升级为 `SELECT FOR UPDATE` 语句，对数据行加悲观锁。
```
Comment (MyonKeminta) [inline pessimistic-transaction.md:112]: 有点担心表述可能造成误解，是不是最好直接说 `SELECT LOCK IN SHARE MODE` 会进行等同于 `SELECT FOR UPDATE` 的行为，而不是说启用 `SELECT LOCK IN SHARE MODE` 的功能比较好？

Comment (MyonKeminta) [inline pessimistic-transaction.md:112]:
```suggestion
    TiDB 不支持共享锁，但提供 `SELECT LOCK IN SHARE MODE` 语法的兼容。默认配置下，`SELECT LOCK IN SHARE MODE` 语句的效果和没有加锁是一样的，不会阻塞其他事务的读写。

    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_upgrade`](/system-variables.md#tidb_enable_shared_lock_upgrade-span-classversion-mark从-v830-版本开始引入span) 系统变量，使  `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意此时加的锁并不实际支持共享；加锁行为与  `SELECT FOR UPDATE` 一致， 实际加的是排他锁。当希望  `SELECT LOCK IN SHARE MODE`  语法可以与写入相互阻塞、避免读期间数据被并行的写入事务修改时，可考虑启用该变量。
```

How about this?

File: system-variables.md [hunk 10]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`, 当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+- 如果该变量和 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 同时开启，该变量会对 `SELECT LOCK IN SHARE MODE` 语句生效。
```
Comment (MyonKeminta) [inline system-variables.md:2302]:
```suggestion
- 无论 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 是否开启，启用该变量都会对 `SELECT LOCK IN SHARE MODE` 语句生效。
```

**Comment by cfzjywxk** [comment]
@joey-yez PTAL again

File: pessimistic-transaction.md [hunk 11]
```diff
@@ -107,7 +107,11 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。
+    
+    可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁是一样的，不会阻塞其他事务的读写。
+
+    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_upgrade`](/system-variables.md#tidb_enable_shared_lock_upgrade-span-classversion-mark从-v830-版本开始引入span) 系统变量，使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。当希望兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，可以与写入相互阻塞、避免读期间数据被并行的写入事务修改时，可考虑启用该变量。 该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```
Comment (hfxsd) [inline pessimistic-transaction.md:114]:
```suggestion
    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_upgrade`](/system-variables.md#tidb_enable_shared_lock_upgrade-span-classversion-mark从-v830-版本开始引入span) 系统变量，使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```

File: pessimistic-transaction.md [hunk 12]
```diff
@@ -107,7 +107,11 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。
+    
+    可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁是一样的，不会阻塞其他事务的读写。
+
+    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_upgrade`](/system-variables.md#tidb_enable_shared_lock_upgrade-span-classversion-mark从-v830-版本开始引入span) 系统变量，使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```
Comment (hfxsd) [inline pessimistic-transaction.md:114]:
```suggestion
    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_promotion`](/system-variables.md#tidb_enable_shared_lock_promotion-从-v830-版本开始引入span) 系统变量，使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```

File: system-variables.md [hunk 13]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 可选值：`OFF`，`ON`
 - 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON`，则优先使用缓存中的 Chunk 对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF`，则直接从系统内存中申请 Chunk 对象。
 
+### `tidb_enable_shared_lock_upgrade` <span class="version-mark">从 v8.3.0 版本开始引入</span>
```
Comment (hfxsd) [inline system-variables.md:2294]:
```suggestion
### `tidb_enable_shared_lock_promotion` <span class="version-mark">从 v8.3.0 版本开始引入</span>
```

File: pessimistic-transaction.md [hunk 14]
```diff
@@ -107,7 +107,11 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。
+    
+    可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁是一样的，不会阻塞其他事务的读写。
```
Comment (hfxsd) [inline pessimistic-transaction.md:112]:
```suggestion
    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁是一样的，不会阻塞其他事务的读写。
```

File: system-variables.md [hunk 15]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 - 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON`，则优先使用缓存中的 Chunk 对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF`，则直接从系统内存中申请 Chunk 对象。
 
+### `tidb_enable_shared_lock_promotion` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`, 当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
```
Comment (lilin90) [inline system-variables.md:2301]:
```suggestion
- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`，当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
```

File: pessimistic-transaction.md [hunk 16]
```diff
@@ -107,7 +107,9 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁是一样的，不会阻塞其他事务的读写。
+
+    从 v8.3.0 版本开始，TiDB 支持使用 [`tidb_enable_shared_lock_promotion`](/system-variables.md#tidb_enable_shared_lock_promotion-从-v830-版本开始引入span) 系统变量，使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```
Comment (lilin90) [inline pessimistic-transaction.md:112]:
```suggestion
    从 v8.3.0 版本开始，TiDB 支持通过启用 [`tidb_enable_shared_lock_promotion`](/system-variables.md#tidb_enable_shared_lock_promotion-从-v830-版本开始引入span) 系统变量使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```

**Review by lilin90 — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 17]
```diff
@@ -2291,6 +2291,16 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
 
+### `tidb_enable_shared_lock_promotion` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用共享锁升级为排他锁的功能。TiDB 默认不支持 `SELECT LOCK IN SHARE MODE`，当该变量值为 `ON` 时，TiDB 会尝试将 `SELECT LOCK IN SHARE MODE` 语句升级为 `SELECT FOR UPDATE` 并真正加悲观锁。该变量默认值为 `OFF`，表示不启用共享锁升级为排他锁的功能。
+- 无论 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入span) 是否开启，启用该变量都会对 `SELECT LOCK IN SHARE MODE` 语句生效。
```
Comment (hfxsd) [inline system-variables.md:2302]:
```suggestion
- 无论 [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-span-classversion-mark从-v40-版本开始引入) 是否开启，启用该变量都会对 `SELECT LOCK IN SHARE MODE` 语句生效。
```

File: pessimistic-transaction.md [hunk 18]
```diff
@@ -107,7 +107,9 @@ BEGIN /*T! PESSIMISTIC */;
 
 2. TiDB 不支持 `SELECT LOCK IN SHARE MODE`。
 
-    使用这个语句执行的时候，效果和没有加锁是一样的，不会阻塞其他事务的读写。
+    TiDB 默认不支持 `SELECT LOCK IN SHARE MODE` 语法。可以通过启用 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 来兼容 `SELECT LOCK IN SHARE MODE` 语法。此时，该语法的效果和没有加锁一样，不会阻塞其他事务的读写。
+
+    从 v8.3.0 版本开始，TiDB 支持通过启用 [`tidb_enable_shared_lock_promotion`](/system-variables.md#tidb_enable_shared_lock_promotion-从-v830-版本开始引入span) 系统变量使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```
Comment (hfxsd) [inline pessimistic-transaction.md:112]:
```suggestion
    从 v8.3.0 版本开始，TiDB 支持通过启用 [`tidb_enable_shared_lock_promotion`](/system-variables.md#tidb_enable_shared_lock_promotion-从-v830-版本开始引入) 系统变量使 `SELECT LOCK IN SHARE MODE` 语句产生加锁行为。但需要注意，此时加的锁并不是真正的共享锁，而是与 `SELECT FOR UPDATE` 一致，实际加的是排他锁。如果你需要兼容 `SELECT LOCK IN SHARE MODE` 语法的同时，希望与写入相互阻塞、避免读期间数据被并行的写入事务修改，可考虑启用该变量。该变量无论 [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) 配置如何都会生效。
```

**Pull request merged**